### ISSUE TITLE
Extensions support

### ISSUE THREAD

**Issue by Armaldio** [getferdi/ferdi#113]
In my browser env, I'm using some useful extensions to enhance services.

It would be very great to support loading these extensions (or at least some of them)

Electron already support some chrome extensions APIs and apps like [Station](https://community.getstation.com/t/how-to-request-an-app-or-an-extension/49), or [Shift](https://support.tryshift.com/kb/article/122--which-chrome-extensions-does-shift-support/) already support this feature.
There is also [this issue for Franz](https://github.com/meetfranz/franz/issues/373)

You can also use extensions for supporting [Dark Theme](https://github.com/getferdi/ferdi/issues/71), [AdBlocker](https://github.com/getferdi/ferdi/issues/69) and many more!

For more information, please see:

- https://github.com/sentialx/electron-extensions
- https://github.com/getstation/electron-chrome-extension
- https://github.com/getstation/repackage-chrome-extension-for-electron
- https://electronjs.org/docs/api/browser-window#browserwindowaddextensionpath

**Comment by rctneil** [comment]
+1 for extension support from me. I'd love to be able to use 1Password.

**Comment by chadberg** [comment]
Another +1. Would love to integrate lastpass.

**Comment by teamtmedia** [comment]
+1 Dashlane

**Comment by vantezzen** [comment]
Related: https://github.com/electron/electron/issues/19447
Electron seems to be getting native support for basic extensions in Electron >= 9. We'll have to wait until that is released and until we have upgraded to it though.

**Comment by kris7t** [comment]
Not stale, in fact, with the release of Electron 9 with extension support, this issue is more relevant than ever

**Comment by vantezzen** [comment]
Yes, we need make Ferdi compatible with Electron 9 first (see https://github.com/orgs/getferdi/projects/3). I'm really looking forward to seeing Ferdi support Extensions.

**Comment by dwehrmann** [comment]
+1 for passbolt (https://www.passbolt.com/) – that would be the missing puzzle piece for me

**Comment by obvionaoe** [comment]
Was gonna create an issue about this, glad to see it's already being worked on!\
I would like to ask for support for the [Stylus extension](https://add0n.com/stylus.html) that would make theming services a lot more customizable, thanks

**Comment by FingerlessGlov3s** [comment]
+1 for me too, would allow me to add Mailvelope, which can be used for many web mail providers, Roundcube, GMail etc.

**Comment by ilikenwf** [comment]
Just something like tampermonkey would be nice too, there's a firefox addon for discord, for example, that somehow reduces battery drain a bit by disabling the discord typing indicators...

Being able to block some of the API urls would help as well, I think that some, especially with discord, are more or less tracking cruft.

**Comment by arielaco** [comment]
+1 for me and I wanted to bump it. Super important feature.

**Comment by Coderdude112** [comment]
+1 Can't wait for this to be released

**Comment by 0xBABA** [comment]
+1 I am really missing this on Gmail for extensions like mixmax

**Comment by SirLouen** [comment]
Same here, I feel this is the last piece for the definitive app given the fact that is built over Electron, which makes this swifter like Wavebox does.

**Comment by kytwb** [comment]
I put up a $50 bounty on this issue ([via bountysource](https://www.bountysource.com/issues/81833962-extensions-support)) to support the work already done by @vantezzen via https://github.com/getferdi/ferdi/pull/822.

**Comment by SithVicious** [comment]
+1 for ad-blockers

**Comment by DarPe07** [comment]
+1 for chrome extensions

**Comment by Knap-Online** [comment]
+1 LastPass and an ad blocker would be awesome!!!

**Comment by milesmcclane** [comment]
Looking forward to the possibility of extensions support also.

**Comment by Potherca** [comment]
With all due respect, but could we stop adding `+1` comments? It just needlessly notifies anyone subscribed to this issue.

To show your support/desire for this issue, give the original post a :+1: and subscribe to be alerted of any updates!

**Comment by vraravam** [comment]
remove staleness

**Comment by warshell** [comment]
Reposting my thoughts: the best would be .CRX extensions, but it would be too much, scripts are cool as they are like extensions. The idea is to implement a userscripts (like greasemonkey, tampermonkey etc )  This would really expand a possibility of software to a new level

**Comment by vraravam** [comment]
@warshell - the idea is good. We are looking for contributors - either to review and/or to help test. As noted by @kytwb in https://github.com/getferdi/ferdi/issues/113#issuecomment-719445447 - we need some contributor to own this up and take it forward.

**Comment by Cull-Methi** [comment]
Just put up an extra $100 on the [bounty](https://www.bountysource.com/issues/81833962-extensions-support) with a 3mo expiration to hopefully speed this along :)

**Comment by avielc** [comment]
Adding my note after exploring Fredi a little more (it seems like it's growth in on turbo right now. ) 
Wish extensions can be introduced here too. something like 1Password and such (which apparently also moved to Electron) would be epic addition. 
Thanks you to all the contributors\bounty boosters and those that gave the idea too!

**Comment by Cull-Methi** [comment]
FYI It looks like two anonymous backers put an additional $200 on the bounty, bringing the total up to $350

**Comment by mbnoimi** [comment]
+1

**Comment by Doumor** [comment]
+1
